source/acquaman/SGM/SGMXASScanConfiguration.cpp

```diff
@@ -94,9 +94,9 @@ AMScanConfigurationView* SGMXASScanConfiguration::createView(){
 }
 
 QString SGMXASScanConfiguration::detailedDescription() const{
-	double exitSlit;
-	double grating;
-	double harmonic;
+	double exitSlit = 0;
+	double grating = 0;
+	double harmonic = 0;
```

Comment:
Thanks @iainworkman for pointing this out. I have the same question when I was working on this. But if you take a look at the codes L346 `SGMBeamline.cpp`, you will find that we guarantee that we have `grating_`, `harmonic_`, and `exitSlitGap_` initialized.